frontend/src/components/common/Pagination.tsx

```diff
@@ -50,15 +50,28 @@ const Pagination = ({
       return;
     }
     setUserPageNum(newUserPageNum);
-    if (!Number.isNaN(newUserPageNum) && newUserPageNum !== pageNum) {
-      getRecords(newUserPageNum);
-    }
   };
+  
+  // Only fetch records if a valid page num is present AND the page num has changed
+  const fetchRecords = () => {
+    if (!Number.isNaN(userPageNum) && userPageNum !== pageNum) {
+      getRecords(userPageNum);
+    }
+  }
 
-  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+  // Treat the enter key as an alt method of triggering onBlur (lose focus)
+  const handleKeyUp = (event: any) => {
+    if (event.keyCode === 13) {
+      event.target.blur();
+    }
+  }
+
+  const handleBlur = () => {
     if (Number.isNaN(userPageNum)) {
       setUserPageNum(pageNum);
+      return;
     }
+    fetchRecords()
   };
```

Comment:
This control flow is a little weird, so I thought I'd comment it to explain:

Two ways to trigger a fetch:
- Exiting focus from the textbox (Clicking outside of it) ; **handleBlur**
- Hitting the enter key ; **handleKeyUp**

handleBlur:
- Checks if number is present ; if not, reset to the previous number and terminate
- If a valid number is present, initiate a fetch

handleKeyUp:
- Checks if key-up (triggered once a user "stops" pressing a key) is == 13 (the key code for the `ENTER` key) ; if true, then this will force a `blur` event (triggering the `handleBlur` function)